node/tests/env0-deploy-utils.spec.js

```diff
@@ -78,11 +75,107 @@ describe("env0-deploy-utils", () => {
           isSensitive
       );
 
-      expect(callApiMock).toHaveBeenCalledWith(
+      expect(mockCallApi).toHaveBeenCalledWith(
           'post',
           'configuration',
           { data: expect.objectContaining({ isSensitive } ) }
       );
     })
   });
+
+  describe('wait for environment', () => {
+    it.each`
+    status
+    ${'CREATED'}
+    ${'INACTIVE'}
+    ${'ACTIVE'}
+    ${'FAILED'}
+    ${'TIMEOUT'}
+    `('should call api client once', async ({ status }) => {
+      mockCallApi.mockResolvedValue({ status });
+
+      await deployUtils.waitForEnvironment(mockEnvironmentId);
+
+      expect(mockCallApi).toBeCalledWith('get', `environments/${mockEnvironmentId}`,);
+      expect(mockCallApi).toBeCalledTimes(1);
+    })
+
+    it('should call api client twice', async () => {
+      mockCallApi.mockResolvedValueOnce({ status: 'TEST' });
+      mockCallApi.mockResolvedValueOnce({ status: 'ACTIVE' });
+
+      await deployUtils.waitForEnvironment(mockEnvironmentId);
+
+      expect(mockCallApi).toBeCalledWith('get', `environments/${mockEnvironmentId}`,);
+      expect(mockCallApi).toBeCalledTimes(2);
+    })
+  });
+
+  describe('poll deployment status', () => {
+    it('should call api twice', async () => {
+      // one for polling and one for writing the logs
+      mockCallApi.mockResolvedValueOnce({ status: 'SUCCESS' });
```

Comment:
This is because all methods use the same api client method so when multiple methods are being called the mock order is important.
Anyway, that test was problematic because it checked the wrong thing.
f12f4f0